src/grt/grt-vpi.adb

```diff
@@ -894,13 +907,15 @@ package body Grt.Vpi is
                Append_Bin (Ghdl_U64 (V), 32);
             end;
          when Vcd_Bit
-           | Vcd_Bool
-           | Vcd_Bitvector =>
+           | Vcd_Bool =>
+            Append (Buf_Value, Map_Std_B1 (Verilog_Wire_Val (Info).B1));
+         when Vcd_Bitvector =>
             for J in 0 .. Len - 1 loop
                Append (Buf_Value, Map_Std_B1 (Verilog_Wire_Val (Info, J).B1));
             end loop;
-         when Vcd_Stdlogic
-           | Vcd_Stdlogic_Vector =>
+         when Vcd_Stdlogic =>
+            Append (Buf_Value, E8_To_Char (Verilog_Wire_Val (Info).E8));
+         when Vcd_Stdlogic_Vector =>
```

Comment:
These are the changes that fixed accessing the values for generics for std_logic, generic and boolean.  The problem is that Verilog_Wire_Val is not implemented by constant/generic arrays.  Previously boolean, std_logic and bit were using the array version.  I switched to using the non-array one where constant/generic are supported.